### PR TITLE
Extract aligned FASTA from a BAM/SAM file

### LINKED ISSUE #577: Add a script to pull aligned reads out of a SAM/BAM file

### PR DESCRIPTION
**Pull request by terrycojones**
Fixes #577.

### CODE REVIEW

File: bin/sam-to-fasta-alignment.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+     help=('If given, the names of all reference sequences will be '
+           'printed to standard output and the program will exit.'))
+
+parser.add_argument(
+     '--listReferenceInsertions', default=False, action='store_true',
+     help=('If given, information about reference sequence insertions will be '
+           'printed to standard error. These correspond to "I" CIGAR '
+           'operations that for the match would require inserting query bases '
+           'into the reference. Because we cannot change the reference (in '
+           'fact we typically do not have the reference in the SAM/BAM file, '
```
Comment (bamueh) [inline bin/sam-to-fasta-alignment.py]: I think you're missing a ')'

File: bin/sam-to-fasta-alignment.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+           'we cut the insert bases out of the aligned query and save the '
```
Comment (bamueh) [inline bin/sam-to-fasta-alignment.py]: Shouldn't this be `inserted bases`?

File: bin/sam-to-fasta-alignment.py (hunk 1 above)
Comment (terrycojones) [reply bin/sam-to-fasta-alignment.py]: Fixed, thanks.

File: bin/sam-to-fasta-alignment.py (hunk 2 above)
Comment (terrycojones) [reply bin/sam-to-fasta-alignment.py]: Yes, fixed too.

**Pull request merged**